packages/back-end/src/models/MetricModel.ts

```diff
@@ -143,7 +150,7 @@ export async function insertMetrics(
 }
 
 export async function deleteMetricById(
-  metric: MetricInterface,
+  metric: LegacyMetricInterface | MetricInterface,
```

Comment:
Because the MetricModel that is used to load metrics before passing them to `deleteMetricById` is `LegacyMetricInterface` which has optional capping and window settings fields.

Reply:
Isn't it using `getMetricById`?  That should always return a migrated `MetricInteface` and not have legacy fields.

Reply:
No, see: https://github.com/growthbook/growthbook/pull/2058/files#diff-f5fef485552f5862881a3d54b3701b0491def49e5cbbc49bc6bf9872ef9d42b0R188

Reply:
Oh, in that case we should keep this function only accepting `MetricInterface` and change that caller to do `toInterface` first.